designer/client/inline-conditions.js

```diff
@@ -0,0 +1,422 @@
+  constructor (props) {
+    super(props)
+
+    const { path } = this.props
+
+    this.state = {
+      conditions: new ConditionsModel(),
+      fields: this.fieldsForPath(path)
+    }
+  }
+
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    if (this.props.path !== prevProps.path) {
+      const fields = this.fieldsForPath(this.props.path)
+
+      const { condition } = this.state
+
+      const newCondition = fields[condition?.field] ? this.state.condition : {}
+      this.setState({
+        conditions: new ConditionsModel(),
+        condition: newCondition,
+        fields: fields
+      })
+    }
+  }
+
+  fieldsForPath (path) {
+    const { data } = this.props
+    return data.inputsAccessibleAt(path)
+      .map(input => ({
+        label: input.title,
+        name: input.name,
+        type: input.type,
+        values: (data.listFor(input)??{}).items
+      }))
+      .reduce((obj, item) => {
+        obj[item.name] = item
+        return obj
+      }, {})
+  }
+
+  onClickAddItem = () => {
+    this.setState({
+      condition: {}
+    })
+  }
+
+  onChangeCoordinator = e => {
+    const input = e.target
+
+    if (input.value && input.value.trim() !== '') {
+      const copy = clone(this.state.condition??{})
+      copy.coordinator = input.value
+      this.setState({
+        condition: copy
+      })
+    } else {
+      this.setState({
+        condition: null
+      })
+    }
+  }
+
+  onClickFinalize = () => {
+    const { conditions, condition } = this.state
+    if (this.state.editing || this.state.editing === 0) {
+      this.setState({
+        conditions: conditions.replace(this.state.editing, new Condition(Field.from(condition.field), condition.operator, Value.from(condition.value), condition.coordinator)),
+        condition: null,
+        editView: true,
+        editing: null
+      })
+    } else {
+      this.setState({
+        conditions: conditions.add(new Condition(Field.from(condition.field), condition.operator, Value.from(condition.value), condition.coordinator)),
+        condition: null
+      })
+    }
+  }
+
+  onChangeField = e => {
+    const input = e.target
+    const fieldName = input.value
+
+    const { condition } = this.state
+
+    const currentField = condition.field?.name
+    const currentOperator = condition.operator
+
+    this._updateCondition(condition, c => {
+      if (fieldName) {
+        if (currentField && this.state.fields[currentField].values !== this.state.fields[fieldName].values) {
+          delete c.value
+        }
+        if (currentOperator && !conditionalOperators.getConditionals(fieldName)[currentOperator]) {
+          delete c.operator
+        }
+        c.field = new Field(fieldName, this.state.fields[fieldName].label)
+      } else {
+        delete c.field
+        delete c.operator
+        delete c.value
+      }
+    })
+  }
+
+  _updateCondition (condition, updates) {
+    const copy = clone(condition)
+    updates(copy)
```

Comment:
It just felt like when passing data that we rely on back to the parent we shouldn't allow any manipulation the parent might do to corrupt our state - especially since it could, potentially, leave the data in a non-processable state